rust/nasl-interpreter/src/built_in_functions/ssh.rs

```diff
@@ -0,0 +1,2107 @@
+    match session.set_option(option) {
+        Ok(_) => (),
+        Err(e) => {
+            FunctionErrorKind::Diagnostic(
+                format!(
+                    "Function {} (calling internal function {}): Failed to disable known_hosts: {}",
+                    "func", "nasl_ssh_connect", e
+                ),
+                Some(NaslValue::Null),
+            );
+        }
+    };
+
+    if !key_type.is_empty() {
+        let option = SshOption::PublicKeyAcceptedTypes(key_type.to_owned());
+        match session.set_option(option) {
+            Ok(_) => (),
+            Err(e) => {
+                return Err(FunctionErrorKind::Diagnostic(
+                    format!(
+                        "Function {} (calling internal function {}): Failed to set SSH key type '{}': {}", "func", "nasl_ssh_connect", key_type, e),
+                    Some(NaslValue::Null)
+                ));
+            }
+        };
+    }
+
+    if !csciphers.is_empty() {
+        let option = SshOption::CiphersCS(csciphers.to_owned());
+        match session.set_option(option) {
+            Ok(_) => (),
+            Err(e) => {
+                return Err(FunctionErrorKind::Diagnostic(
+                    format!(
+                        "Function {} (calling internal function {}): Failed to set SSH client to server ciphers '{}': {}", "func", "nasl_ssh_connect", csciphers, e),
+                    Some(NaslValue::Null)
+                ));
+            }
+        };
+    }
+
+    if !scciphers.is_empty() {
+        let option = SshOption::CiphersSC(scciphers.to_owned());
+        match session.set_option(option) {
+            Ok(_) => (),
+            Err(e) => {
+                return Err(FunctionErrorKind::Diagnostic(
+                    format!(
+                        "Function {} (calling internal function {}): Failed to set SSH server to client ciphers '{}': {}", "func", "nasl_ssh_connect", scciphers, e),
+                    Some(NaslValue::Null)
+                ));
+            }
+        };
+    }
+
+    let valid_ports = 1..65535;
+    if valid_ports.contains(&port) {
+        let option = SshOption::Port(port);
+        match session.set_option(option) {
+            Ok(_) => (),
+            Err(e) => {
+                return Err(FunctionErrorKind::Diagnostic(
+                    format!(
+                        "Function {} (calling internal function {}) called from {}: Failed to set SSH port '{}': {}", "func", "nasl_ssh_connect", "key", port, e),
+                    Some(NaslValue::Null),
+                ));
+            }
+        };
+    }
+
+    let mut forced_sock = -1;
+    if sock > 0 {
+        // This is a fake raw socket.
+        // TODO: implement openvas_get_socket_from_connection()
+        let my_sock = UdpSocket::bind("127.0.0.1:0").unwrap();
+        let option = SshOption::Socket(my_sock.as_raw_fd());
+
+        if verbose > 0 {
+            ctx.logger().info(&format!(
```

Comment:
because this is use an ENV variable to switch it on, different from the common debug level. It will only be available if it is set via, which also turns the libssh debug log on     
```
let verbose = env::var("OPENVAS_LIBSSH_DEBUG")
```